DataDog/integrations-core
# Add dbstrict option to limit queries to specified databases

### What does this PR do?

Adds dbstrict parameter to restrict to only use dbname(s) listed. Currently dbname can only accept one name, but this could be adapted to set a single instance listing only the dbs that a customer wants to monitor. The issue here is that currently if a client wants to limit monitoring to only certain databases, the pg_database_size() forces them to open permissions to all databases. Furthermore, if a customer sets up several instances we'll essentially run the same query for each one which isn't great from a efficiency standpoint. 

Adding a dbstrict will allow customers to either restrict to a single db for all metrics or reduce down to just those that they configure. In the future it would be good to also allow for multiple dbnames since each postgres instance can contain multiple databases. 
 
### Motivation


[Zendesk Ticket](https://datadog.zendesk.com/agent/tickets/444659)
[JIRA Ticket](https://datadoghq.atlassian.net/browse/AGENT-5810)

### Additional Notes

Already tested with ddev and a equivalent setup and this does indeed limit this down to only truly monitoring a single database.

Steps for testing with ddev:

1) launched the normal ddev with my dev branch
2) created a new database in the postgres instance
     CREATE DATABASE db2;
3) revoked permissions for the datadog role
     REVOKE ALL PRIVILEGES ON db2 FROM datadog;
4) enabled new param dbstrict on my single instance that included the database datadog_test as the database
5) the check started running and only reported size info for the datadog_test database instead of all 3.

### Review checklist (to be filled by reviewers)
